test/parallel/test-util-inspect.js

```diff
@@ -1151,4 +1151,21 @@ if (typeof Symbol !== 'undefined') {
 }
 
 assert.doesNotThrow(() => util.inspect(process));
+
+// @@toStringTag
+assert.strictEqual(util.inspect({ [Symbol.toStringTag]: 'a' }),
+                   'Object [a] { [Symbol(Symbol.toStringTag)]: \'a\' }');
+
+class Foo {
+  constructor() {
+    this.foo = 'bar';
+  }
+
+  get [Symbol.toStringTag]() {
+    return this.foo;
+  }
+}
+
+assert.strictEqual(util.inspect(new Foo()), 'Foo [bar] { foo: \'bar\' }');
```

Comment:
Would be glad to see inspection works for objects of a subclass of Foo as well. E.g.

```js
assert.strictEqual(
  util.inspect(new (class extends Foo {})()),
  'Foo [bar] { foo: \'bar\' }');
assert.strictEqual(
  util.inspect(Object.create(Object.create(Foo.prototype), {
    foo: { value: 'bar' }
  })),
  'Foo [bar] { foo: \'bar\' }');
```